### ISSUE TITLE
Draft: Exception during shapes editing

### ISSUE THREAD

**Issue by m1cha1** [FreeCAD/FreeCAD#14006]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

While editing basic draft shapes (line, rectangle, polyline etc.) I have an exception:
<Exception> TopoShapeExpansion.cpp(2032): Invalid shape name EditNode0

Steps:
1. Draw rectangle
2. Edit rectangle-change the size or position
3. Exception

Similiar sitaution [here](https://forum.freecad.org/viewtopic.php?p=759354).

### Full version info

```shell
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37302 (Git)
Build type: Release
Branch: main
Hash: 0e24e121eb3e5708c380596e0a9fd583befac977
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: Polish/Poland (pl_PL)
Installed mods: 
  * 3DfindIT 1.2.0
  * Curves 0.6.35
  * Design456 0.0.1
  * fasteners 0.5.16
  * lattice2 1.0.0
  * Manipulator 1.5.7
  * pcb 6.2023.1
  * Rocket 3.3.0
  * sheetmetal 0.4.13
  * workfeature
  * workfeature-macro
```

### Subproject(s) affected?

Draft

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
@bgbsww is this related to toponaming?

**Comment by bgbsww** [comment]
I don't think so, as "EditNode0" is illegal, and very unrelated to anything toponaming.  The referenced similar situation was apparently solved by #14017.  Totally willing to be wrong, but it doesn't seem to fit the toponaming space.

**Comment by maxwxyz** [comment]
@Roy-043 FYI

**Comment by bgbsww** [comment]
I take that back.  I was unable to replicate when running under a recent build with the TNP fix enabled.  But I was able to replicate with the TNP fix off.  That implies an inadvertent change to the code that would be related.

**Comment by m1cha1** [comment]
Fixed in: Version: 0.22.0dev.37358 (Git)
